wordpress-mobile/WordPress-Android
# Remove locale aware activity

As a follow up to #21521, this PR removes `LocaleAwareActivity`. There are a lot of files changed in this PR, but this was because we extended `LocaleAwareActivity` in so many places. In this PR, I changed them all to extend `AppCompatActivity`.

Testing steps are the same as the prior PR.

Test 1:

- Run the app on a device using Android 13 or later
- Go to App Settings and tap "Interface language"
- Note that the system dialog for "App Language" appears instead of our own
- Verify changing languages works

Test 2:

- Perform the above with an older device and verify our own language dialog appears and changing languages works

**Note:** There's still quite a bit of cleanup to do related to locale handling (unnecessary functions, etc.), but I wanted to keep this PR dedicated to removing `LocaleAwareActivity`. Cleanup is being addressed in [this draft PR](https://github.com/wordpress-mobile/WordPress-Android/pull/21527).
